include/seqan3/argument_parser/validators.hpp

```diff
@@ -375,13 +374,16 @@ class file_validator_base
         auto drop_less_ext = tmp_str | views::drop(1);
 
         // Compares the extensions in lower case.
-        auto cmp_lambda = [&] (std::string const & cmp)
+        auto case_insensitive_equal_to = [&] (std::string const & ext)
         {
-            return std::ranges::equal(cmp | views::to_lower, drop_less_ext | views::to_lower);
+            return std::ranges::equal(ext, drop_less_ext, [] (char const chr1, char const chr2)
+                   {
+                       return std::tolower(chr1) == std::tolower(chr2);
```

Comment:
not that performance would be an issue here but we do have our own seqan3::to_lower predicates in core char_operations. It also does not pull the ranges...

Reply:
I know, but this is one step in decoupling argument parser from seqan3.

Reply:
yajk I forgot.